### DISCUSSION TITLE
Return Youtube Dislike for Safari (iOS + macOS)

### DISCUSSION THREAD

**Discussion by rishikdev** [Anarios/return-youtube-dislike#935] [Q&A]
# Update
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful!

> [!NOTE]
> This application/extension **requires** macOS 14.6 or later, iOS 15.0 or later, iPadOS 15.0 or later, and visionOS 1.0 or later.

> [!TIP]
> If the above hyperlink does not work, here is the link to the extension: https://apple.co/3B7zfFr

# Old Discussion Post
Hi, I have made an extension for Safari (iOS and macOS) using [userscript](https://raw.githubusercontent.com/Anarios/return-youtube-dislike/main/Extensions/UserScript/Return%20Youtube%20Dislike.user.js), and It is working as expected. Can I publish it to the App Store for everyone to use? It is going to be a free extension, and I am not going to charge anything for it in anyway.

Edit: For people who want to install the RYTD application on their iOS or macOS devices, you can do so using this TestFlight link: https://testflight.apple.com/join/EWGBcZT6. I could publish the application to the App Store, but I would need the owner's permission to do it as I do not own the program constituting the application.

**Comment by ddpp1** [comment]
Can you publish for ios safari too?

**Reply by rishikdev** [reply]
I have built the application for iOS and macOS. I am waiting for the builds to be approved to be able to distribute them on TestFlight!

**Reply by ddpp1** [reply]
That sounds fantastic! Thank you!

**Reply by rishikdev** [reply]
https://testflight.apple.com/join/EWGBcZT6

**Reply by ddpp1** [reply]
I installed TestFlight for the first time and will test it out. It’s working perfectly so far. Great work.

**Reply by ddpp1** [reply]
It stopped working and was removed from the TestFlight app. Is the extension out now?

**Reply by rishikdev** [reply]
The current build has expired. I will try to create a new build and upload it to TestFlight soon.

**Reply by ddpp1** [reply]
That would be awesome. It works perfectly. 

Do you have a way to follow the project?

**Reply by rishikdev** [reply]
Here is the TestFlight link: https://testflight.apple.com/join/EWGBcZT6

You can Watch or Star this repository to follow it. However, I don't think that there is a separate repository for Safari.

**Reply by ddpp1** [reply]
King. Thank you so much for doing this and for letting us know. 

For the record, I’m sure it’s complicated to do, but I would pay for this in extension form.

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by JannesHch** [comment]
Very cool of you to do that! I was just looking for an extension like this on the AppStore, but only found one for 3€, which is insane in my opinion. I considered making an extension myself but good to hear that someone else is already on it! Could you let me join your TestFlight too? It would be appreciated :)

**Reply by rishikdev** [reply]
Here is the TestFlight link: https://testflight.apple.com/join/EWGBcZT6

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by ikekreham** [comment]
It's working perfectly ! And a huge "Tank you" for sharing !
When do you think the extension will be available in the Mac AppStore ?

**Reply by rishikdev** [reply]
I don't know when the extension will be made available. I did not create it; I simply ported it to Xcode. So, without the owner's permission, I cannot upload the extension to the App Store for ethical reasons. The best that I can do is make it available through TestFlight.

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by elixss** [comment]
I'd appreciate if you would accept new testers:) would love to try it out!

**Reply by rishikdev** [reply]
I have redeployed it to TestFlight. You should be able to download it from there. Here is the link: https://testflight.apple.com/join/EWGBcZT6

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by tlarro** [comment]
My TestFlight has expired too, is it possible to get a new link? It worked perfectly

**Reply by rishikdev** [reply]
Here is the link just in case you didn’t receive an email or a notification on TestFlight: https://testflight.apple.com/join/EWGBcZT6

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by tlarro** [comment]
Thank you very much ! 😄

Le dim. 31 mars 2024 à 08:01, Rishik Dev ***@***.***> a
écrit :

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by TheCornworld** [comment]
I would also like a beta link

**Reply by rishikdev** [reply]
Sorry for the delay. Here is the link: https://testflight.apple.com/join/EWGBcZT6

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by deKato** [comment]
Any plans to put it to AppStore? (TestFlight is outdated)

**Reply by rishikdev** [reply]
I don't know when the extension will be made available. I did not create it; I simply ported it to Xcode. So, without the owner's permission, I cannot upload the extension to the App Store for ethical reasons. The best that I can do is make it available through TestFlight. I have updated the build in TestFlight, so you should be able to download it. Here is the link: https://testflight.apple.com/join/EWGBcZT6

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by jedai47** [comment]
would love the link @rishikdev

**Reply by rishikdev** [reply]
Here you go: https://testflight.apple.com/join/EWGBcZT6

**Reply by jedai47** [reply]
I have 'Unable to open link'

**Reply by jedai47** [reply]
u can send me the code on 'laasmiwalid(at)hotmail(dot)com' thanks a lot @rishikdev

**Reply by rishikdev** [reply]
Do you have the TestFlight application on your device?

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by moon-cacke** [comment]
Just added the script with shortcuts as well : https://github.com/Anarios/return-youtube-dislike/issues/1098

**Comment by Azzhoe** [comment]
The link keeps saying the beta isn't accepting anymore users. How can I join please?

**Reply by rishikdev** [reply]
The build has expired; maybe that is the reason why you cannot join the group. I will try to rebuild the project and upload it to TestFlight once it is ready.

**Reply by ansuff** [reply]
I’m also interested in getting it. Please keep us posted when you upload it! Thanks!

**Reply by rishikdev** [reply]
I have made the extension available for everyone - it can be found on the App Store [here](https://apple.co/3B7zfFr)! The extension is available for macOS, iOS, iPadOS, and visionOS I hope you will find it useful! See the updated post description for more details.

**Comment by ovojake** [comment]
Hey are you still working on this? wondering how difficult it would be to add back the little bar underneath the likes/dislikes. it's been a while since I've used RYD on pc but iirc it had this feature :)

**Reply by rishikdev** [reply]
It depends on the original developers. I just took their script and published it as a macOS/iOS Safari extension.

**Reply by ovojake** [reply]
I double checked and can confirm it already exists on the official extension. As well as the little menu to make some of those changes. Are these features not possible to port over to safari- with the current code?

**Comment by Julmik647** [comment]
As you used userscript, add option for users to use these userscript. It would be helpful on iPhone.
https://github.com/Dwyriel/Greasyfork-Scripts/tree/main/ReturnYoutubeDislikeOnMobile
